### PR TITLE
Added Huddle Features

### PR DESCRIPTION
**Pull request by noranda**
# Added Huddle Features
- Added `add-to-huddle-modal` component to handle manually adding patients to huddles. If a huddle date is selected where a huddle already exists for at least one other patient, populates with a read-only leader. This approach ensures that only one huddle group can be scheduled per day.
- Added `pikaday` [datepicker](https://github.com/dbushell/Pikaday) to handle picking dates for huddles.
- Added styles to datepicker to indicate scheduled and reviewed huddles for a given patient.
- Added next huddle date for each patient in patient list. Shows as blank if no huddle scheduled.
- Added patient view display of next scheduled huddle date with option to mark as reviewed.
- Added patient view scheduling feature to view scheduled huddle details, mark a scheduled huddle as reviewed, or schedule a new huddle for a patient.
- Added `review-patient-modal` component to handle marking patients in huddle as reviewed.
- Added `is-today-or-after` helper to help with scheduling.
- Added `patient-huddles` helper to help with displaying patient huddles.
- Added `has-stylesheet` mixin to add and remove stylesheets for the datepickers.
- Added `huddle` model to handle huddles and `huddle-patient` model to handle patients in huddles. Includes `toFhirJson` function to format data correctly for FHIR while keeping the frontend data simple to use.
- Added styles to new huddle features and modals.
##### TODO:
- [ ] Implement edit feature for huddles.
- [ ] Handle case where patients not marked as reviewed should be rescheduled to next huddle.
##### Video:

https://vimeo.com/161503227

### CODE REVIEW

File: app/templates/components/add-to-huddle-modal.hbs [hunk 1]
```diff
@@ -0,0 +1,62 @@
+{{#bootstrap-modal title="Add to Huddle" onOpen=attrs.onOpen onClose=attrs.onClose}}
+  {{#if isLoading}}
+    <div class="modal-body modal-body-spinner">
+      {{ember-spinner config='small'}}
+    </div>
+  {{else}}
+    {{#if patientInExistingHuddle}}
+      <div class="alert alert-danger" style="position:relative;margin:0 0 20px;top:0;left:0;width:auto;">
```
Comment (schreiaj) [inline app/templates/components/add-to-huddle-modal.hbs]: Do we want to do this as an in line style?

File: app/templates/components/patient-search/huddle-list.hbs [hunk 2]
```diff
@@ -12,13 +12,23 @@
...
+            class="css-checkbox"
+            checked={{active}}
+            onclick={{action 'toggle'}}>
           <label for="huddleList1" class="css-label css-label-box checkbox-label">
             Geriatrics Huddle
           </label>
-          <i class="fa fa-edit pull-right"></i>
-          <div class="date-selector" style="padding-left:40px;font-size:0.8em;">
-            {{date-format nextMonday 'll'}} <span class="fa fa-chevron-down"></span>
+          <div class="date-selector" id="huddleFilterSelector" style="padding-left:40px;font-size:0.8em;cursor:pointer;">
```
Comment (schreiaj) [inline app/templates/components/patient-search/huddle-list.hbs]: Another inline style.

File: app/templates/components/patient-viewer.hbs [hunk 3]
```diff
@@ -46,7 +46,53 @@
...
+                  {{/if}}
+
+                  <div class="meta">
+                    {{#if selectedScheduleHuddle}}
+                      Geriatrics Huddle<br>
+                      Leader: {{selectedScheduleHuddle.displayLeader}}<br>
+                      {{selectedScheduleHuddlePatient.reasonText}}
+                      {{#if selectedScheduleHuddlePatient.reviewed}}
+                        <br>
+                        Reviewed on {{moment-format selectedScheduleHuddlePatient.reviewed 'MMM D, YYYY'}}
```
Comment (schreiaj) [inline app/templates/components/patient-viewer.hbs:86]: Do we want to extract this into a single helper so we are consistent across displaying dates everywhere?

File: app/templates/components/review-patient-modal.hbs [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+      </div>
+
+      <div class="form-group">
+        <div class="form-control-static row">
+          <div class="col-sm-6">
+            <label>Huddle Date:</label>
+          </div>
+
+          <div class="col-sm-6 form-value">
+            {{moment-format huddle.date 'MMM DD, YYYY'}}
```
Comment (schreiaj) [inline app/templates/components/review-patient-modal.hbs:35]: Date display again.

File: app/templates/components/review-patient-modal.hbs [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+          </div>
+
+          <div class="col-sm-6 form-value">
+            <div class="input-addon left-addon">
+              <i class="fa fa-calendar-o fa-fw left-addon-icon"></i>
+              {{pikaday-input
+                id="reviewDate"
+                class="form-control form-input-calendar"
+                value=reviewDate
+                format="MMMM Do YYYY"
```
Comment (schreiaj) [inline app/templates/components/review-patient-modal.hbs:65]: Example - this format string is different than the other ones.

File: package.json [hunk 6]
```diff
@@ -52,6 +53,8 @@
     "ember-fhir-adapter": "0.0.12",
     "ember-load-initializers": "0.5.1",
     "ember-modal-dialog": "0.8.3",
+    "ember-moment": "6.1.0",
```
Comment (schreiaj) [inline package.json:56]: If we are installing this via NPM do we need to still have it in Bower?

File: app/components/patient-search/huddle-list.js [hunk 7]
```diff
@@ -1,23 +1,121 @@
...
+    }
+  }),
+
+  maxDate: computed('huddles.@each.date', {
+    get() {
+      let huddles = this.get('huddles');
+      if (huddles.length === 0) {
+        return null;
+      }
+      return get(huddles.sortBy('date').objectAt(huddles.length - 1), 'date');
```
Comment (schreiaj) [inline app/components/patient-search/huddle-list.js]: `get(huddles.sortBy('date').objectAt(huddles.length - 1), 'date')` could be replaced with `get(huddles.sortBy('date').get('lastObject.date')` as it's a little clearer what the intent is.

File: app/components/patient-search/huddle-list.js [hunk 8]
```diff
@@ -1,23 +1,121 @@
...
+      setDefaultDate,
+      minDate: this.get('minDate'),
+      maxDate: this.get('maxDate'),
+      disableDayFn: (date) => {
+        let huddleDateMap = this.get('huddleDateMap');
+        return huddleDateMap[moment(date).format('YYYY-MM-DD')] === undefined;
+      },
+      onSelect: (date) => {
+        let huddleDateMap = this.get('huddleDateMap');
+        this.attrs.selectHuddle(huddleDateMap[moment(date).format('YYYY-MM-DD')]);
```
Comment (schreiaj) [inline app/components/patient-search/huddle-list.js:86]: I don't think this is displayed to users anywhere but it's yet another date format.

File: app/components/add-to-huddle-modal.js [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+      this.set('formSaving', true);
+
+      let huddle = this.get('existingHuddle');
+      let newHuddle = false;
+
+      if (huddle == null) {
+        huddle = Huddle.create({
+          date: this.get('huddleDate'),
+          leader: `Practitioner/${this.get('huddleLeader')}`,
+          name: moment(this.get('huddleDate')).format('MMMM D, YYYY') // TODO: change this
```
Comment (schreiaj) [inline app/components/add-to-huddle-modal.js:109]: Again on the standardized dates.

**Comment by schreiaj** [comment]
@noranda If you can remove the inline CSS this is good to merge. 

I talked it over with @cmoesel and I think doing a quick check of the different date formats we use in the app is in the cards just to maintain consistency. I'll add a story in Pivotal for it.

File: app/templates/components/patient-viewer.hbs (hunk 3 above)
Comment (noranda) [inline app/templates/components/patient-viewer.hbs:86]: While it's usually a good rule-of-thumb to keep the date display UI the same, this won't always be the case. We can always revisit this once we have the entire app fleshed out to see where we can optimize with helpers.

File: app/templates/components/review-patient-modal.hbs (hunk 4 above)
Comment (noranda) [inline app/templates/components/review-patient-modal.hbs:35]: `moment-format` is already a helper in `ember-moment`.

File: app/templates/components/patient-search/huddle-list.hbs (hunk 2 above)
Comment (noranda) [inline app/templates/components/patient-search/huddle-list.hbs]: Fixed.

File: app/templates/components/add-to-huddle-modal.hbs (hunk 1 above)
Comment (noranda) [inline app/templates/components/add-to-huddle-modal.hbs]: Fixed.

File: app/templates/components/review-patient-modal.hbs (hunk 5 above)
Comment (noranda) [inline app/templates/components/review-patient-modal.hbs:65]: The format here is specific for `pikaday` calendar date selections.

File: package.json (hunk 6 above)
Comment (noranda) [inline package.json:56]: Yes, we still need it in Bower. `ember-moment` installs `ember-cli-moment-shim` in `package.json`, and `ember-cli-moment-shim` imports `moment.js` from our bower folder. Neither of them package `moment.js`, so we still need to include it.

File: app/components/patient-search/huddle-list.js (hunk 7 above)
Comment (noranda) [inline app/components/patient-search/huddle-list.js]: Changed to `return huddles.sortBy('date').get('lastObject.date');`.

File: app/components/patient-search/huddle-list.js (hunk 8 above)
Comment (noranda) [inline app/components/patient-search/huddle-list.js:86]: Again, this is specific to `pikaday`.

File: app/components/add-to-huddle-modal.js (hunk 9 above)
Comment (noranda) [inline app/components/add-to-huddle-modal.js:109]: See comment above.

**Pull request merged**